rust/kernel/thread.rs

```diff
@@ -0,0 +1,508 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A kernel thread (kthread).
+//!
+//! This module allows Rust code to create/wakup/stop a kernel thread.
+
+use crate::c_types;
+use crate::error::{ptr_to_result, Error, KernelResult};
+use crate::{bindings, cstr, CStr};
+
+use alloc::boxed::Box;
+use core::ops::FnOnce;
+
+extern "C" {
+    #[allow(improper_ctypes)]
+    fn rust_helper_get_task_struct(task: *mut bindings::task_struct);
+    #[allow(improper_ctypes)]
+    fn rust_helper_put_task_struct(task: *mut bindings::task_struct);
+}
+
+/// Generates a bridge function from Rust to C ABI.
+///
+/// `func` should be a `fn(arg: arg_type) -> KernelResult<()>` where arg_type
+/// is the same size as `*mut c_types::c_void` ([`core::intrinsics::transmute`]
+/// checks that at the compile time.
+#[macro_export]
+macro_rules! bridge {
```

Comment:
```suggestion
macro_rules! thread_bridge {
```

And `#[doc(hidden)]` please.

Reply:
I'm considering to remove the macro entirely.

Reply:
Done in the updated version. the macro way is gone ;-)